src/GraphQL/Validation/TypeInfo.cs

```diff
@@ -28,43 +28,52 @@ public TypeInfo(ISchema schema)
             _schema = schema;
         }
 
-        /// <summary>
-        /// Returns an ancestor of the current node.
-        /// </summary>
-        /// <param name="index">Index of the ancestor; 0 for the node itself, 1 for the direct ancestor and so on.</param>
-        public INode GetAncestor(int index)
+        private static T PeekElement<T>(Stack<T> from, int index)
         {
-            var e = _ancestorStack.GetEnumerator();
-
-            int i = index;
-            do
+            if (index == 0)
             {
-                _ = e.MoveNext();
+                return from.Count > 0 ? from.Peek() : default;
             }
-            while (i-- > 0);
+            else
+            {
+                var e = from.GetEnumerator();
 
-            return e.Current; // throws if index is out of range
+                int i = index;
+                do
+                {
+                    _ = e.MoveNext();
+                }
+                while (i-- > 0);
+
+                return e.Current; // throws if index is out of range
```

Comment:
It should throw, but the behavior of `IEnumerable.Current` in that case is "undefined".  We should check the return value of `MoveNext`.

Reply:
fixed